src/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,322 @@
+/**
+* Description   - Opportunity Trigger Handler
+* Created by    - Sagar Nirwal
+* Created Date  - 01/08/2022
+* Revision log  -  V_1.0 - Created  - Sagar Nirwal - 01/08/2022
+*                  V_1.1 - Modified - Sagar NIrwal - 01/08/2022 -  Details what you modified
+**/
+public with sharing class OpportunityTriggerHandler {
+    /**   Description :   Method to create Child records if opportunity is created
+    *     Created By  :   Sagar Nirwal
+    *     Arguments   :   List<Opportunity>,Map<Id, Opportunity>
+    *     Return      :   void
+    *     Revision log:   V_1.0 - Created  - Sagar Nirwal - 01/08/2022
+    *                     V_1.1 - Modified - Sagar Nirwal - 01/08/2022 - Details what you modified
+    **/
+    public static void automobileSensor(List<Opportunity>  newOppList, Map<Id, Opportunity> oldMapOpp){
+        List<SObject> listRecordsToInsert = new List<SObject>();
+        Set<String> sensorofSet = new Set<String>();
+        for(Opportunity opp : newOppList){
+            if(oldMapOpp == null || opp.StageName != oldMapOpp.get(opp.Id).StageName){
```

Comment:
You have handled insert and update conditions but you are calling this method from insert only.